js/fairdesk.js

```diff
@@ -0,0 +1,1450 @@
+            this.parseNumber (this.safeString (ohlcv, 'high')),
+            this.parseNumber (this.safeString (ohlcv, 'low')),
+            this.parseNumber (this.safeString (ohlcv, 'close')),
+            this.parseNumber (this.safeString (ohlcv, 'volume')),
+        ];
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        const request = {
+            'interval': timeframe,
+        };
+        let startTime = 0;
+        let endTime = 0;
+        const duration = this.parseTimeframe (timeframe);
+        const now = this.seconds ();
+        if (since !== undefined) {
+            if (limit === undefined) {
+                limit = 1000; // max 1000
+            }
+            since = parseInt (since / 1000);
+            startTime = since;
+            // time ranges ending in the future are not accepted
+            // https://github.com/ccxt/ccxt/issues/8050
+            endTime = Math.min (now, this.sum (since, duration * limit));
+        } else if (limit !== undefined) {
+            limit = Math.min (limit, 1000);
+            startTime = now - duration * this.sum (limit, 1);
+            endTime = now;
+        } else {
+            throw new ArgumentsRequired (this.id + ' fetchOHLCV() requires a since argument, or a limit argument, or both');
+        }
+        request['from'] = startTime * 1000;
+        request['to'] = endTime * 1000;
+        await this.loadMarkets ();
+        const symbolId = this.parseSymbol (symbol);
+        const market = this.market (symbolId);
+        request['symbol'] = market['id'];
+        const response = await this.publicGetMdKline (this.extend (request, params));
+        // {
+        //     "status": 0,
+        //     "error": "OK",
+        //     "data": [    {
+        //         "openTime": 1651382700000,
+        //         "intervalId": "5m",
+        //         "closeTime": 1651382999999,
+        //         "open": 38076.000000,
+        //         "close": 38095.000000,
+        //         "high": 38098.000000,
+        //         "low": 38056.500000,
+        //         "avgPrice": 38066.500000,
+        //         "volume": 0.067000,
+        //         "quoteVolume": 2551.822500,
+        //         "symbol": "btcusdt",
+        //         "closed": true,
+        //         "numTrades": 15
+        //     },]
+        // }
+        const rows = this.safeValue (response, 'data', {});
+        return this.parseOHLCVs (rows, market, timeframe, since, limit);
+    }
+
+    parseTicker (ticker, market = undefined) {
+        const marketId = this.safeString (ticker, 'symbol');
+        market = this.safeMarket (marketId, market);
+        const symbol = market['symbol'];
+        const timestamp = this.safeInteger (ticker, 'timestamp');
+        const last = this.safeNumber (ticker, 'close');
+        const open = this.safeNumber (ticker, 'open');
+        const change = last - open;
```

Comment:
We should pass strings to `safeTicker` here, the values of `change` and `percentage` will be calculated there.

Also, please, check this section: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings